spec.md

```diff
@@ -0,0 +1,238 @@
+- Ignore the contents and validation of the config.json file.
+
+## Optional: Defining Artifact Publisher Manifests
+
+Artifact authors may wish to publish their types as [well known types][def-well-known-types] for registry operators to consume. By publishing an `artifactManifest.json`, registry operators can parse information for validation, registry display and search results.
+
+An `artifactType.json` file is defined using the [following schema](artifactTypes/artifactTypeSchema.0.1.json).
+
+- **`mediaType`** *string*  
+  This REQUIRED property uniquely identifies the artifact for computer consumption. It may be owned by an org or a company and MUST be globally unique and versioned.  
+  The format of `mediaType` MUST use the following format:  
+  `application/vnd.`[org|company]`.`[objectType]`.`[optionalSubType]`.config.`[version]`+json`
+- **`title`** *string-string map*  
+  This REQUIRED property must have at least one value, representing the name of the type displayed for human consumption. The title may be displayed in a repository listing, or registry tooling.  
+  Title is a collection of localized strings, indexed with [ISO Language Codes][iso-lang-codes].
+  - **`locale`** *string*  
+    2-2 language codes representing the country and locale. [ISO Language Codes][iso-lang-codes]
+  - **`title`** *string*  
+Localized title. The max length MUST not exceed 30 characters and MUST not encode formatting characters.
+- **`description`** *string-string map*  
+    This REQUIRED property must have at least one value, providing a short description of the type for human consumption. The description may be displayed in repository listings or registry tooling.  
+    Description is a collection of localized strings, indexed with [ISO Language Codes][iso-lang-codes].  
+  - **`locale`** *string*  
+    2-2 language codes representing the country and locale. [ISO Language Codes][iso-lang-codes]
+  - **`description`** *string*  
+Localized description. The max length MUST not exceed 255 characters and MUST not encode formatting characters.
+- **`moreInfoUrl`** *url*  
+This OPTIONAL property provides additional details, intended for consumers of the type. This is most often a marketing & awareness overview page.
+- **`specUrl`** *url*  
+This OPTIONAL property references a spec, providing additional details on the type.
+- **`tools`** *string-string map*  
+  This OPTIONAL property provides a collection of tools that may be used with artifact type. The property is intended for end users to find more info on how to find and install related tools. Registry operators MAY provide links to the tools in their repository listing.
+  - **`url`** *url*  
+    This REQUIRED property links to a page where users can download the tool. The URL MAY be a direct link to a download URL, or a link to documentation for how to download the tool.
+  - **`title`** *string-string map*  
```

Comment:
Seems like you'd want a `description` map as well so you could differentiate tools (authoring tools, validation tools, runtime/consuming tools, etc.).

Reply:
I was hoping to minimize the duplication a toolset would need to maintain in publishing its info. By listing urls and title, i was thinking that was enough. As tools, i imagine the descriptive text would quickly evolve as they test their messaging. Having to resubmit a localized PR to OCI Artifacts each time would be onerous. 
I could add it if we feel it’s really needed. I’d like to see how registry operators use this info to figure out what info is needed.

Reply:
> I could add it if we feel it’s really needed. I’d like to see how registry operators use this info to figure out what info is needed.

That's reasonable, and you can easily add properties like that without backwards-compat concerns.

Thinking that over, one thing I don't see in this PR so far is versioning for the `artifactType.json` payload.  Are artifact authors supposed to upload those to registries?  In which case, what would they set for `Content-Type`?  Or are they supposed to serve them somewhere for registry operators to pull down and drop into a configuration directory?  In which case, how will the registry determine the version of the artifact definition?

Reply:
`artifactType.json` could be pushed to this repo, for well-known types, which registry operators could import. 
That said, while we've discussed a distribution-spec api for importing these files, I'm deferring this conversation for the moment, focusing on the what's in the payload, before we discuss how operators could import them.
Versioning is a good question. In [this example](https://github.com/SteveLasker/artifacts/tree/master/artifactTypes/vnd.oci.image.1), I've versioned the `artifactType.json` by placing it in a sub folder. The premise is the file name doesn't change, rather the folder it's in. Thoughts?

Reply:
I see you're asking for a categorization, not just a description. Some types will likely have more than others. So, I'm going to defer this for now to see how this plays out.
I did add [Spec for how to upload `artifactType.json` and discover supported types #14](https://github.com/opencontainers/artifacts/issues/14) for tracking.